src/snyk/advisor/services/AdvisorDisposable.ts

```diff
@@ -0,0 +1,78 @@
+import { Subscription } from 'rxjs';
+import { IVSCodeLanguages } from '../../common/vscode/languages';
+import { getModules, getSupportedLanguage, isValidModuleName } from '../../common/vscode/parsing';
+import { ThemeColorAdapter } from '../../common/vscode/theme';
+import { Disposable, Language, TextDocument } from '../../common/vscode/types';
+import { IVSCodeWindow } from '../../common/vscode/window';
+import { ModuleVulnerabilityCountProvider } from '../../snykOss/services/vulnerabilityCount/vulnerabilityCountProvider';
+import EditorDecorator from '../editor/EditorDecorator';
+import AdvisorService from './AdvisorService';
+
+export class AdvisorScoreDisposable implements Disposable {
+  protected disposables: Disposable[] = [];
+  protected advisorScanFinishedSubscription: Subscription;
+
+  private readonly editorDecorator: EditorDecorator;
+
+  constructor(
+    private readonly window: IVSCodeWindow,
+    private readonly languages: IVSCodeLanguages,
+    private readonly advisorService: AdvisorService,
+    private readonly vulnerabilityCountProvider: ModuleVulnerabilityCountProvider,
+  ) {
+    this.editorDecorator = new EditorDecorator(window, languages, new ThemeColorAdapter());
+  }
+
+  activate(): boolean {
+    this.advisorScanFinishedSubscription = this.advisorService.scanFinished$.subscribe(() => {
+      this.processActiveEditor();
+    });
+    return false;
+  }
+
+  processActiveEditor(): void {
+    const activeEditor = this.window.getActiveTextEditor();
+    if (activeEditor) {
+      this.processFile(activeEditor.document);
+    }
+  }
+
+  processFile(document: TextDocument): boolean {
+    if (!document) {
+      return false;
+    }
+
+    const { fileName, languageId } = document;
+    const supportedLanguage = getSupportedLanguage(fileName, languageId);
+    if (supportedLanguage !== Language.PJSON) {
+      return false;
+    }
+    const scores = this.advisorService.getScoresResult();
+    if (scores?.length) {
+      if (supportedLanguage) {
+        const modules = getModules(fileName, document.getText(), supportedLanguage).filter(isValidModuleName);
+        const promises = modules
+          .map(module => this.vulnerabilityCountProvider.getVulnerabilityCount(module, supportedLanguage))
```

Comment:
Why do you need this map? Vulnerability Count doesn't relate to advisor.

Reply:
Because we only show advisor score for packages that has vulnerabilities.

Reply:
Is it the case in IntelliJ Advisor integration as well? I didn't see the feature plan and tbh this wouldn't be my expectation 🤔

Reply:
That's the requirements that I got from Oren - @snyk/advisor

Reply:
Please verify if we're aligned on that requirement with IntelliJ plugin.